hack/lib/util/golang.sh

```diff
@@ -11,8 +11,7 @@ function os::golang::verify_go_version() {
 	if [[ "${go_version[2]}" < "${OS_REQUIRED_GO_VERSION}" ]]; then
 		os::log::info "Detected go version: ${go_version[*]}."
 		if [[ -z "${PERMISSIVE_GO:-}" ]]; then
-			os::log::warning "Please install Go version ${OS_REQUIRED_GO_VERSION} or use PERMISSIVE_GO=y to bypass this check."
-			return 0
```

Comment:
intentional? Then also delete the return below.

Reply:
Yes, this needs to fail, when you want this to fallback you need to use `PERMISSIVE_GO` envvar. Not sure which return you're talking about.